src/main/java/Problem1/ValidParentheses.java

```diff
@@ -4,8 +4,40 @@
 
     // Do not change signature (function name, parameters, return type)
     public static boolean isValid(String str) {
-        // homework
-        // !!! must use ArrayStack or LinkedListStack from problem 1
-        return false; // place holder
+        //Uses ArrayStack
+        if (str == null || str.length() == 0) {
+            return true;
+        }
+        Stack<Character> stk= new ArrayStack<>(str.length());
+        char c;
+        for (int i = 0; i < str.length(); i++) {
+            c = str.charAt(i);
+            if (c == '(' || c == '{' || c == '[' || c == ')' || c == '}' || c == ']') {
+                switch (c) {
+                    case '(':
+                    case '{':
+                    case '[':
+                        stk.push(c);
+                        break;
+                    case ')':
+                    case '}':
+                    case ']':
+                        if (stk.peek() != null) {
+                            char last = stk.peek();
+                            if (c == ')' && last == '(' || c == '}' && last == '{' || c == ']' && last == '[') {
+                                stk.pop();
+                                if (i == str.length() - 1) { //check if it is the last one
+                                    return true;
+                                }
+                            } else {
+                                return false;
+                            }
+                        }
+                }
+            } else {
+                return false;
+            }
+        }
+        return false;
```

Comment:
logic is good. code can be refactored for better readability. also there's duplication here and there. something like this would be easier to read:

```java
   public static boolean isValid(String str) {
        if (str == null || str.isEmpty() || str.trim().length() == 0) {
            return true;
        }

        Stack<Character> stack = new LinkedListStack<>();

        for (int i = 0; i < str.length(); i++) {
            char c = str.charAt(i);
            Character left = stack.peek();
            if (left == null || !characterMatch(left, c)) {
                stack.push(c);
                continue;
            }
            stack.pop();
        }
        return stack.size() == 0;
    }

    private static boolean characterMatch(char left, char right) {
        switch (left) {
            case '(':
                return right == ')';
            case '{':
                return right == '}';
            case '[':
                return right == ']';
        }
        return false;
    }
```